snuba/admin/google.py

```diff
@@ -1,49 +1,71 @@
+from abc import ABC
+from typing import Optional
 from urllib.parse import urlencode
 
 import structlog
-from googleapiclient.discovery import Resource
+from googleapiclient.discovery import Resource, build
 
 logger = structlog.get_logger().bind(module=__name__)
 
 
-def get_group_id(service: Resource, group_email: str) -> str:
-    try:
-        query_params = f"groupKey.id={group_email}"
-        request = service.groups().lookup()
-        request.uri += "&" + query_params
-        response = request.execute()
-        if "error" in response:
-            logger.exception(
-                f"An HTTP error occured when fetching group id for email {group_email}.",
-                google_api_error=response["error"],
-            )
+class GoogleAPIWrapper(ABC):
+    """
+    An abstract class defining an interface for google cloud APIs.
+    """
+
+
+class CloudIdentityAPI:
+    """
+    An interface for the Google Cloud Identity API.
+    """
+
+    def __init__(self) -> None:
+        try:
+            self.service: Resource = build("cloudidentity", "v1")
+        except Exception as e:
+            logger.exception(e)
```

Comment:
Small nit: I might set a `self.initialized = False` here and `self.initialized = True` on success. Then in the other functions you can check `self.initialized` instead of catching Exceptions.

Reply:
I've added this flag, but I think it's a good idea to catch exceptions here, so that they're silently caught and the end user will not necessarily know about it. I think this might be a good place to send this exception to sentry as well, so we can be alerted if something silently breaks. @evanh What do you think?